tests/ci/integration/run_curl_integration.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+
+# SYS_ROOT
+#  |
+#  - SRC_ROOT(aws-lc)
+#  |
+#  - SCRATCH_FOLDER
+#    |
+#    - curl
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+#    - CURL_BUILD_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SYS_ROOT}/"CURL_BUILD_ROOT"
+CURL_SRC_FOLDER="${SCRATCH_FOLDER}/curl"
+CURL_BUILD_FOLDER="${SCRATCH_FOLDER}/curl/build"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${CURL_SRC_FOLDER}/aws-lc-install"
+
+mkdir -p ${SCRATCH_FOLDER}
+rm -rf ${SCRATCH_FOLDER}/*
+cd ${SCRATCH_FOLDER}
+
+function curl_build() {
+  autoreconf -fi
+  ./configure --enable-warnings --enable-werror --with-openssl=${AWS_LC_INSTALL_FOLDER}
+  make -j ${NUM_CPU_THREADS}
+  make -j ${NUM_CPU_THREADS} examples
+}
+
+function curl_run_tests() {
+  make -j ${NUM_CPU_THREADS} tests
+  make test-ci
+  cd ${SCRATCH_FOLDER}
+}
+
+# Get latest curl version.
+git clone https://github.com/curl/curl.git ${CURL_SRC_FOLDER}
```

Comment:
Does curl make a guarantee that their main branch will always be buildable/release quality? If not, is that something we should be concerned about in regards to our CI?

Reply:
Not much of a guarantee since they do periodic releases, but they run a similar CI dimension against a version of AWS-LC in their CI as well. This helps us have some faith that we'll always be buildable with the tip of main.
* https://github.com/curl/curl/blob/master/.github/workflows/awslc.yml

We've been following a pattern of building with main for most of these integrations. If it's deemed too unstable, we can always pin to a specific version instead :).